### ISSUE TITLE
Curry the `withState` so that we can compose/pipeline the addons

### ISSUE THREAD

**Issue by wmzy** [dump247/storybook-state#6]
The code is hard to read:
```js
storiesOf('myStories', module)
  .add('myFun', withState({}, withInfo(`
  description or documentation about my component, supports markdown
  `)(myFun)));
```

With the curried,we could write it like this:
```js
storiesOf('myStories', module)
  .add('myFun', _.pipe(
    withInfo(`
  description or documentation about my component, supports markdown
    `),
    withState({}),
  )(myFun));
```
or
```js
storiesOf('myStories', module)
  .add('myFun', myFun
    |> withInfo(`
  description or documentation about my component, supports markdown
    `)
    |> withState({})
  );
```

**Comment by dump247** [comment]
Unfortunately, it won't work the way you propose. At least, not directly. The store is passed as an argument to the story function to use in rendering the component.

Your first example would have to be:

```javascript
storiesOf('myStories', module)
  .add('myFun', withState({}, store => withInfo(`
  description or documentation about my component, supports markdown
  `)(() => <myFun {...store.state}/>)));
```

Let me know if you have any suggestions for how to improve on that. Somehow you have to get the store to the function that actually renders the component. This [other storybook state component](https://github.com/Sambego/storybook-state) gets around it by making the store global declared at a higher scope. I prefer to have the store limited to the scope of the story, particularly when you have a bunch of stories that use state.

**Comment by wmzy** [comment]
There is an example for curry: https://github.com/wmzy/formsy-antd/blob/ffe03ec479a6dec581406ca78dbe76bac2c5217f/stories/horizontal-login.js#L61-L64

I think of it is a good idea to export a curried `withState`.

**Comment by dump247** [comment]
@wmzy I made the change, although slightly different than what you suggested. I added the `store` to the storybook context object, which is an undocumented object that gets passed down the storybook function chain. Now it should work with the [info addon](https://github.com/storybooks/storybook/tree/master/addons/info) in any order.

Example:
```javascript
const myFun = ({ store }) => <MyComponent {...store.state}/>;

storiesOf('myStories', module)
  .add('myFun', myFun
    |> withInfo(`
  description or documentation about my component, supports markdown
    `)
    |> withState({})
  );
```

**Comment by dump247** [comment]
Resolved in v1.4.0
